tasks/change-granule-collection-s3/src/update_cmr_file_collection.ts

```diff
@@ -0,0 +1,108 @@
+import { AssertionError } from 'assert';
+import { s3PutObject } from '@cumulus/aws-client/S3';
+import {
+  DistributionBucketMap,
+  generateEcho10XMLString,
+  isECHO10Filename,
+  isUMMGFilename,
+  setECHO10Collection,
+  updateEcho10XMLMetadataObject,
+  setUMMGCollection,
+  updateUMMGMetadataObject,
+} from '@cumulus/cmrjs/cmr-utils';
+import { ApiFile, ApiGranuleRecord } from '@cumulus/types';
+import { log } from '@cumulus/common';
+import { ValidationError } from '@cumulus/errors';
+import {
+  ValidApiFile,
+  ValidGranuleRecord,
+} from './types';
+
+export function apiFileIsValid(file: Omit<ApiFile, 'granuleId'> | ApiFile): file is ValidApiFile {
+  if (file.bucket === undefined || file.key === undefined || file.fileName === undefined) {
```

Comment:
We should *probably* just calculate fileName upstream of this method (this is fine if we do) if it's missing - fileName is optional in the DB and ApiFile object types.